gh-65210: Add const qualifiers in PyArg_VaParseTupleAndKeywords() (#105958)
python/cpython: Doc/c-api/arg.rst

   Returns true on success; on failure, it returns false and raises the
   appropriate exception.

   .. note::

      The *keywords* parameter declaration is :c:expr:`char * const *` in C and
      :c:expr:`const char * const *` in C++.

>       This can be overridden by defining the macro :c:macro:`PY_CXX_CONST`

Reviewer: Please document this new macro somewhere with `.. c:macro:: PY_CXX_CONST`.

That will fix this warning and the one in `3.13.rst` and the CI.

Author: I do not know how to document it better than I did. I will appreciate your suggestions. For now I just removed the role.

Reviewer: @CAM-Gerlach Any suggestions?

Reviewer 2: @serhiy-storchaka To properly document a public entity (macro, function, class, method, etc), you can use the the appropriate directive—in this case, `.. c:macro::` as @hugovk mentioned. This could be something like (placed, perhaps, at the top or bottom of the `API Functions` subsection):

```rst
.. macro:: PY_CXX_CONST

   The value to be inserted, if any, before :c:expr:`char * const *`
   in the *keywords* parameter declaration of
   :c:func:`PyArg_ParseTupleAndKeywords`
   and :c:func:`PyArg_VaParseTupleAndKeywords`.
   Default empty for C and ``const`` for C++ (:c:expr:`const char * const *`).
   To override, define it to the desired value before including :file:`Python.h`.

   .. versionadded:: 3.13
```

You can then simplify this `note` accordingly, per my other suggestion (in a separate comment).

>       This can be overridden by defining the macro :c:macro:`PY_CXX_CONST`
>       before including :file:`Python.h` as ``const`` for the latter and as
>       empty value for the former.

Reviewer 2: ```suggestion
      This can be overridden with the :c:macro:`PY_CXX_CONST` macro.
```

Refer to the newly-added macro's documentation instead of needing to repeat these details here.
